### ISSUE TITLE
Find a substitute for dockerhub

### ISSUE THREAD

**Issue by stweil** [OCR-D/core#997]
Announced today: " Docker is sunsetting Free Team organizations". That means https://hub.docker.com/u/ocrd "will be suspended on April 14, 2023 (11:59 pm UTC). Upgrade your subscription before April 14, 2023 to continue accessing your organization."

The subscription is not cheap, so I am afraid that OCR-D cannot stay there. Therefore a substitute is required, and much documentation and some code (scripts) has to be updated.

**Comment by stweil** [comment]
Possible container repository alternatives:

- GitHub, (currently) free
- self hosted Gitlab
- self hosted Gitea (first test at UB Mannheim was successful)
- ... and many more

**Comment by bertsky** [comment]
> * GitHub, (currently) free

I recommend this for now. It's fairly [easy to setup](https://github.com/slub/ocrd_manager/blob/main/.github/workflows/publish.yml) on Github Actions and has a nice UI integration (_packages_ on the right, replacing Dockerhub, with versions directly tied to commits). The only change on the user side would be the additional qualifier on the image name (i.e. `ghcr.io/ocrd/...` instead of `ocrd/...`).

**Comment by stweil** [comment]
The additional qualifier is also needed for running container images with `podman` which would have required `docker.io/ocrd/...`. For a self hosted Gitea it would be something like `code.bib.uni-mannheim.de/ocrd/...`.

I also think that using GitHub as the new container repository is the easiest way to go. My only concern is that Microsoft won't offer that service for free forever.

**Comment by kba** [comment]
Agreed, let's switch over to the GiHub container registry. @lena-hinrichsen https://github.com/OCR-D/ocrd-website/issues/317 we should fix the documentation accordingly.

**Comment by stweil** [comment]
... and copy all existing images from docker.io to ghcr.io? That's a lot of work, but I think that would be helpful.

**Comment by kba** [comment]
> Maybe there already exist scripts which can copy all container images of an organization.

https://gist.github.com/FriendlyUser/b25d11002a46364a99b9c0f442eff6c9 looks like a simple (though time- and space-intensive) way to do that.

**Comment by kba** [comment]
> > Maybe there already exist scripts which can copy all container images of an organization.
> 
> https://gist.github.com/FriendlyUser/b25d11002a46364a99b9c0f442eff6c9 looks like a simple (though time- and space-intensive) way to do that.

[This guide](https://blog.alexellis.io/docker-is-deleting-open-source-images/) recommends the [`crane`](https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane.md) tool which can copy from registries without pulling/pushing.

**Comment by stweil** [comment]
The guide reminds me of an additional problem which probably will occur: base containers which are used for OCR-D might also move away from docker.io. So let's be prepared when that happens.

**Comment by stweil** [comment]
"After listening to the concerns of the community, we’ve decided to reverse course, and are no longer sunsetting the Free Team plan."

**Comment by bertsky** [comment]
Also just [picked](https://www.docker.com/blog/no-longer-sunsetting-the-free-team-plan/) it up.

That's good, and we got lucky this time, but IMO we should continue the new efforts at diversifying our deployment options. Does not harm to also use GHCR or custom Gitlab servers when available, and write configs and makefiles and documentation a little more flexibly.
